src/libsync/syncengine.cpp

```diff
@@ -1048,7 +1050,9 @@ void SyncEngine::slotItemCompleted(const SyncFileItem &item, const PropagatorJob
 
 void SyncEngine::slotFinished(bool success)
 {
-    _anotherSyncNeeded = _anotherSyncNeeded || _propagator->_anotherSyncNeeded;
+    if (_propagator->_anotherSyncNeeded && _anotherSyncNeeded == NoFollowUpSync) {
```

Comment:
Hm. I think I did it this way because I considered WithDelay to be stronger - it will always have an effect, while ImmediateFollowUp only works a couple of times before it is ignored. It's a situation that won't usually come up. I'd keep it as is and add a comment. Agreed?